cashu/wallet/cli/cli.py

```diff
@@ -81,31 +108,40 @@ def cli(ctx: Context, host: str, walletname: str):
     ctx.ensure_object(dict)
     ctx.obj["HOST"] = host or settings.mint_url
     ctx.obj["WALLET_NAME"] = walletname
-    wallet = Wallet(
-        ctx.obj["HOST"], os.path.join(settings.cashu_dir, walletname), name=walletname
-    )
+
+    db_path = os.path.join(settings.cashu_dir, walletname)
+    # loop = asyncio.new_event_loop()
+
+    # if the command is "restore" we don't want to ask the user for a mnemonic
+    # otherwise it will create a mnemonic and store it in the database
+    if ctx.invoked_subcommand == "restore":
+        wallet = await Wallet.with_db(
+            ctx.obj["HOST"], db_path, name=walletname, skip_private_key=True
+        )
+    else:
+        # # we need to run the migrations before we load the wallet for the first time
+        # # otherwise the wallet will not be able to generate a new private key and store it
+        # loop.run_until_complete(migrate_wallet_db(Database("wallet", db_path)))
```

Comment:
Is this comment also deprecated? Because it looks like the migration is actually not done.

Reply:
Yes, this line was deprecated as well. The comment relates to the new `cls` method `.with_db()` where we need to `skip_private_key` once in cases where the db wasn't migrated yet. Not ideal solution but still a lot better than before. I think this can be cleaned up later.